phonebuddies/DatabaseConnector.py

```diff
@@ -156,7 +160,14 @@ def _find_labeled_row_text(label_text):
         intro = _find_labeled_row_text("Intro")
         topic = _find_labeled_row_text("Topic")
         return EmailInfo(intro, topic)
-    
+
+    def is_email_in_database(self, email):
+        # Get the map of all buddies with their emails as keys
+        buddy_map = self.get_all_buddies_map()
+
+        # Check if the email exists in the buddy_map, considering it case-insensitively
+        return email.lower() in buddy_map
```

Comment:
While this is theoretically useful, I think it's best we only keep the functions we are currently using for now.